contrib/http_util/src/main/java/io/opencensus/contrib/http/AbstractHttpHandler.java

```diff
@@ -66,36 +69,33 @@ private static void putAttributeIfNotEmptyOrNull(Span span, String key, @Nullabl
   }
 
   /**
-   * Instrument an HTTP span after a message is sent.
+   * Instrument an HTTP span after a message is sent. Typically called when last chunk of request or
+   * response is sent.
    *
-   * @param span the span.
-   * @param messageId an id for the message.
-   * @param messageSize the size of the message.
-   * @since 0.18
+   * @param context request specific {@link HttpContext}
+   * @since 0.19
    */
-  // [TODO:rghetia] add it back after 0.18 is released
-  final void handleMessageSent(Span span, long messageId, long messageSize) {
-    checkNotNull(span, "span");
-    if (span.getOptions().contains(Options.RECORD_EVENTS)) {
+  public final void handleMessageSent(HttpContext context) {
+    checkNotNull(context, "context");
+    if (context.span.getOptions().contains(Options.RECORD_EVENTS)) {
       // record compressed size
-      recordMessageEvent(span, messageId, Type.SENT, messageSize, 0L);
+      recordMessageEvent(context.span, context.reqId, Type.SENT, context.sentMessageSize.get(), 0L);
```

Comment:
Seems to me that reqId is not what should be used here. The message id should be a unique id (for each message sent/received in a stream). reqId seems to be constant for the entire request.

So what we need to record is the following:
message sent from client {sent, id = 1, size} - on server {recv, id = 1, size}
message recv by client {recv, id = 1, size} - on server {send, id = 1, size}
message recv by client {recv, id = 2, size} - on server {send, id = 2, size}
message recv by client {recv, id = 3, size} - on server {send, id = 3, size}

Reply:
so the even is not for when the entire request/response content is sent/received but rather the individual chunks. 
What is the significance of the id? How is it used?

Reply:
I have changed it to record the event per message sent/received instead of per request.